posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,442 @@
+
+* The deprecated `@Health` qualifier was removed. Instead, the `@Liveness` or `@Readiness` qualifiers should be used as appropriate in your HealthCheck implementations.
+
+Applications are expected to provide health check procedures by implementing the HealthCheck interface with the `@Liveness` or `@Readiness` annotations. These are used by Open Liberty to verify the Liveness or Readiness of the application, respectively. Add the logic of your health check in the call() method, and return an instance of `HealthCheckResponse` by using the simple `up` and `down` API methods:
+
+[source, java]
+----
+//**Liveness Check**
+@Liveness
+@ApplicationScoped
+public class AppLiveCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.up("my-liveness-check");
+       ...
+     }
+}
+
+//**Readiness Check**
+@Readiness
+@ApplicationScoped
+public class AppReadyCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.named("my-app-readiness").status(isMyAppReady()).build();
+       ...
+     }
+}
+...
+----
+
+To view the status of each health check, access the either the `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
+
+More information:
+
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.0[MicroProfile Health 3.0 Release Page (Javadoc & Spec)]
+
+* link:https://download.eclipse.org/microprofile/microprofile-health-3.0/microprofile-health-spec-3.0.html#release_notes_3_0[Release notes]
+
+[#opentracing-2.0]
+=== New updates in  MicroProfile Open Tracing 2.0
+
+MicroProfile OpenTracing 2.0 can be used to profile and monitor applications built using microservice architecture.
+
+MicroProfile OpenTracing 2.0 has upgraded the OpenTracing API to version 0.33.0.  This allows the
+use of tracing backends and their libraries that are built on OpenTracing API 0.33.0.
+
+==== Try it now
+
+Include the following in the `server.xml`:
+
+[source, xml]
+----
+    <feature>mpOpenTracing-2.0</feature>
+----
+
+Also configure a tracing backend such as Jaeger or Zipkin. +
+For Jaeger, add the following maven dependencies in the application's pom.xml.
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.jaegertracing</groupId>
+    <artifactId>jaeger-client</artifactId>
+    <version>1.2.0</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-api</artifactId>
+    <version>1.7.30</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-jdk14</artifactId>
+    <version>1.7.30</version>
+</dependency>
+----
+
+You can find out more about about configuring Jaeger settings using environment variables by looking
```

Comment:
We also have a guide on enabling distributed tracing using Jaeger [[here](https://openliberty.io/guides/microprofile-opentracing-jaeger.html)]